v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+
+    (void)strcpy(logger_name, start_event_trace_property_data.logger_name);
+
+    // enable our desired provider
+    ULONG enable_trace_result = EnableTraceEx2(trace_session_handle, (LPCGUID)&provider_guid, EVENT_CONTROL_CODE_ENABLE_PROVIDER,
+        trace_level, 0, 0, 0, NULL
+    );
+    POOR_MANS_ASSERT(enable_trace_result == ERROR_SUCCESS);
+
+    return trace_session_handle;
+}
+
+static void stop_trace(TRACEHANDLE trace_session_handle)
+{
+    EVENT_TRACE_PROPERTY_DATA stop_event_trace_property_data = { 0 };
+
+    stop_event_trace_property_data.props.Wnode.BufferSize = sizeof(stop_event_trace_property_data);
+    stop_event_trace_property_data.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+    stop_event_trace_property_data.props.LoggerNameOffset = offsetof(EVENT_TRACE_PROPERTY_DATA, logger_name);
+    stop_event_trace_property_data.props.LogFileNameOffset = 0;
+
+    (void)printf("Stopping trace session %s\r\n", trace_session_name);
+    POOR_MANS_ASSERT(StopTraceA(trace_session_handle, NULL, &stop_event_trace_property_data.props) == ERROR_SUCCESS);
+}
+
+DWORD WINAPI process_trace_thread_func(void* arg)
+{
+    (void)arg;
+
+    // open the trace and process all events
+    ULONG process_trace_result = ProcessTrace(&trace, 1, NULL, NULL);
+    POOR_MANS_ASSERT(process_trace_result == ERROR_SUCCESS);
+
+    return 0;
+}
+
+#define TEST_PARSE_START_DELAY 5000 // ms
+#define TEST_WAITTIME_NO_EVENT 20000 // ms
+
+static void start_parse_events(void)
+{
+    EVENT_TRACE_LOGFILEA log_file = { 0 };
+
+    parsed_event_count = 0;
+
+    log_file.ProcessTraceMode = PROCESS_TRACE_MODE_REAL_TIME | PROCESS_TRACE_MODE_EVENT_RECORD;
+    log_file.EventRecordCallback = event_trace_record_callback;
+    log_file.BufferCallback = event_trace_buffer_callback;
+    log_file.LoggerName = logger_name;
+    log_file.Context = NULL;
+
+    trace = OpenTraceA(&log_file);
+    POOR_MANS_ASSERT(trace != INVALID_PROCESSTRACE_HANDLE);
+
+    DWORD thread_id;
+    process_thread_handle = CreateThread(NULL, 0, process_trace_thread_func, NULL, 0, &thread_id);
+    POOR_MANS_ASSERT(process_thread_handle != NULL);
+
+    // Give the thread some time to start
+    Sleep(TEST_PARSE_START_DELAY);
+}
+
+static void stop_parse_events(void)
+{
+    WaitForSingleObject(process_thread_handle, INFINITE);
+
+    (void)CloseHandle(process_thread_handle);
+    (void)CloseTrace(trace);
+}
+
+static void wait_for_event_count(int32_t expected_event_count)
+{
+    do
+    {
+        int32_t current_parsed_event_count = InterlockedAdd(&parsed_event_count, 0);
+        if (current_parsed_event_count == expected_event_count)
+        {
+            break;
+        }
+
+        (void)WaitOnAddress(&parsed_event_count, &current_parsed_event_count, sizeof(int32_t), INFINITE);
+    } while (1);
+}
+
+static void log_sink_etw_log_with_LOG_LEVEL_ERROR_succeeds(void)
```

Comment:
>log_sink_etw_log_with_LOG_LEVEL_ERROR_succeeds [](http://example.com/codeflow?start=12&length=46)

Looks like this is "special" because it captures also the initial setup "self test" of the logging provider.

(as opposed to log_sink_etw_log_all_levels_when_all_levels_enabled_succeeds).

So that would imply that changing the order in which `log_sink_etw_log_with_LOG_LEVEL_ERROR_succeeds` and `log_sink_etw_log_all_levels_when_all_levels_enabled_succeeds` are written would lead to tests failing.

So I would propose to have a "eat_first_event" with some description about why it is discarded. Had to uncompile lines 376-381

Reply:
I added an extra comment. I wanted to also check this self test event rather than just discard it.,